app/services/cosmos/cosmos.ts

```diff
@@ -1,63 +1,121 @@
-import Cosmos from "@lunie/cosmos-api"
+/* eslint-disable @typescript-eslint/camelcase */
+import {
+  DistributionExtension,
+  QueryClient,
+  setupDistributionExtension,
+  setupStakingExtension,
+  StargateClient,
+  StakingExtension,
+  SigningStargateClient,
+  BroadcastTxResponse,
+  MsgSendEncodeObject,
+  MsgDelegateEncodeObject,
+  MsgUndelegateEncodeObject,
+  MsgWithdrawDelegatorRewardEncodeObject,
+} from "@cosmjs/stargate";
+import { OfflineDirectSigner } from '@cosmjs/proto-signing';
+import { Tendermint34Client } from "@cosmjs/tendermint-rpc";
+import { TextDecoder } from 'text-decoding';
+import BigNumber from "bignumber.js";
 
+import { MsgSend } from 'cosmjs-types/cosmos/bank/v1beta1/tx';
+import {
+  MsgBeginRedelegate,
+  MsgDelegate,
+  MsgUndelegate,
+} from 'cosmjs-types/cosmos/staking/v1beta1/tx'
+import { MsgWithdrawDelegatorReward } from "cosmjs-types/cosmos/distribution/v1beta1/tx";
 import {
-  CosmosAccountResult,
   CosmosCoinResult,
   CosmosDelegation,
   CosmosMessage,
+  CosmosMessageToSign,
+  CosmosRedelegation,
   CosmosRewardsResult,
+  CosmosSigner,
   CosmosUnbondingDelegation,
   CosmosValidator,
 } from "./cosmos.types"
 import {
-  extractCoinFromCosmosCoinList,
-  parseCosmosCoin,
+  convertValidator,
+  convertDecCoin,
+  convertDelegationDelegatorReward,
+  convertDelegationResponse,
+  convertUnbondingDelegation,
+  convertRedelegation,
 } from "./cosmos.utils"
+import { MintExtension, setupMintExtension } from "./mint-query-extension"
 
 /**
  * Cosmos API helper for LikeCoin
  */
 export class CosmosAPI {
-  /**
-   * The Cosmos API client
-   */
-  api: Cosmos
+  defaultGasLimits = {
+    send: 80000,
+    delegate: 160000,
+    transfer: 160000,
```

Comment:
What is different between `send` and `transfer` ?

Reply:
I think the `send` is for `/cosmos.bank.v1beta1.MsgSend`, but the `transfer` is for `/ibc.applications.transfer.v1.MsgTransfer`.
Maybe we don't need `transfer` here.